crates/ide_completion/src/completions/record.rs

```diff
@@ -45,9 +47,102 @@ pub(crate) fn complete_record(acc: &mut Completions, ctx: &CompletionContext) ->
     Some(())
 }
 
+pub(crate) fn complete_record_literal(
+    acc: &mut Completions,
+    ctx: &CompletionContext,
+) -> Option<()> {
+    if let hir::Adt::Struct(strukt) = ctx.expected_type.as_ref()?.as_adt()? {
```

Comment:
We need some filters here, currently we also complete this in pattern positions which we don't wanna do here.

I think 
```rs
if !ctx.expects_expression() {
    return None;
} 

```
should suffice.

This will also get rid of 3 changed pattern tests which are incorrectly triggering the completion currently.